### ISSUE TITLE
How to indicate which projects are public

### ISSUE THREAD

**Issue by leberrigan** [MotusDev/Motus-TO-DO#1112]
From Catherine in #720: 

We need a way for each user to find out which projects are public so they know what they can access.

We talked about a couple ways we could do this, including

-Flagging public projects on the project exploration page: https://sandbox.motus.org/data/projects
-A new R function that returns a list of projects that are public.
-A list on the profile page: https://sandbox.motus.org/data/user/edit
-Others?

We should talk about this on Thursday. For now I've created a new view called vwR_User_Proj that lists all projects a user has access to download data from (using tagme(project)).

**Comment by steffilazerte** [comment]
I'm not available to do this until later in June, but is this something I should give a shot at added? Is the view ready for being accessed by an R function?

**Comment by cjardine-bsc** [comment]
Hi Steffi! 

Yes, this is something we want to add to the R package. We can check with @denislepage to see if there is an API for that view set up. 

Right now the view contains: 

[project_id]
      ,[projectName]
      ,[descriptionShort]
      ,[url]
 
@j-sayers @leberrigan @StuMackenzie  are there any other field you'd like? 

We might want this available on the website as well.

**Comment by leberrigan** [comment]
What about project contact? That may be important to include and should be fine if it's only visible to logged-in users.

**Comment by cjardine-bsc** [comment]
Yeah, I debated about that. Seems iffy to allow users to download email lists through R. So the url links to the project page where contact is listed. What do others think?

**Comment by denislepage** [comment]
Agree. Just a link to the project.

**Comment by steffilazerte** [comment]
Is there an API entry point ready for this?

**Comment by steffilazerte** [comment]
Ping @denislepage, do we have an API entry point?

**Comment by denislepage** [comment]
Doesn't look like we had yet, no, but I just created one now.

The entry point (beta only for now) should be, if you want to test it @steffilazerte:

https://beta.motus.org/api/sgdata/metadata_project_data_for_user

The fields returned should be:

[projectId]
[projectName]
[descriptionShort]
[url]

The results should be based on the user token.

We may want to add a column that provides the reason why the user has access, whether it is because the project is open, or they have permission, but that can wait for Catherine's return.

**Comment by steffilazerte** [comment]
Great thanks, I'll take a look!

**Comment by cjardine-bsc** [comment]
-replace "researchers" with project access here: https://sandbox.motus.org/data/projects

-Add table listing the projects the user has access to on their profile (pulled from [dbo].[vwR_User_Proj]) link each to it's project page (https://sandbox.motus.org/data/project?id=0)

**Comment by cjardine-bsc** [comment]
@steffilazerte  Is there anything else you need from us to add this to the Motus R package?

**Comment by steffilazerte** [comment]
Probably just time 😅 I haven't checked the api point yet.

The only other thing is to confirm the functionality, I assume this would be provided as a new function which would simply return the API values?

**Comment by cjardine-bsc** [comment]
Perfect! That's correct!

**Comment by bcreightonyoung** [comment]
There are two things I want to clarify quick for this issue:

- @leberrigan For the project explore page, which type of "public" access are we meaning in this discussion? My assumption is currently "open data/open summary data/private", i.e. the securable_bin field. The only other project-related fields that I figure would make sense are "is_private_project" and "access_level", but the vast majority of projects have values "false" and "3" for those columns respectively, so the feature would be fairly trivial in those cases.
- There is already a sort of projects table displayed on the user's profile. It is on the /data page on the right panel, "View your profile". Is this what we are looking for? I queried vwR_User_Proj a few times and the result set for each user seems identical to the table that is currently there.

**Comment by cjardine-bsc** [comment]
Heya @bcreightonyoung 

RE: For the project explore page, which type of "public" access are we meaning in this discussion? My assumption is currently "open data/open summary data/private", i.e. the securable_bin field. 

Good question, and correct assumption! You'll use he securable_bin field. 

RE: There is already a sort of projects table displayed on the user's profile. It is on the /data page on the right panel, "View your profile". Is this what we are looking for? I queried vwR_User_Proj a few times and the result set for each user seems identical to the table that is currently there.

Hmm that is true, this area shows places where the user can access metadata management tools. 
![image](https://user-images.githubusercontent.com/22172758/228380333-4cec8ba7-698e-4477-a796-1f20b4b5bd09.png)

What I'd like to make sure we communicate to users is which projects they can access raw data from through R, which is slightly different. I could see that going on the same page, just below the existing table with the title/subtitle

Projects with Data Access

Raw data is accessible through the Motus R Package for the following projects: 

TABLE

**Comment by bcreightonyoung** [comment]
Gotcha!

**Comment by bcreightonyoung** [comment]
@cjardine-bsc this is basically done, with one small obstacle:

![null_securable_bins](https://user-images.githubusercontent.com/124801048/228920209-e7852f54-2a16-4669-9f09-5d2e9985858a.png)

As you can see, quite a lot of projects have a null securable_bin. Maybe this is only a sandbox problem but I'm not sure. For the table on data/projects, under the new column project access, I was just going to display an empty string for the projects with a null securable_bin. Does that sound reasonable?

**Comment by bcreightonyoung** [comment]
Oh good, that works. Thanks!
